### DISCUSSION TITLE
DEBUG: ECR getAuthorizationToken

### DISCUSSION THREAD

**Discussion by Melson95** [renovatebot/renovate#11001] [Request Help]
Issue:

```
DEBUG: ECR getAuthorizationToken error (repository=org/repo)
TRACE: Using Basic auth for docker registry https://***.dkr.ecr.eu-west-2.amazonaws.com (repository=org/repo)
DEBUG: Failed to get authHeaders for getTags lookup (repository=org/repo)
TRACE: Saving cached value (repository=org/repo)
       "namespace": "datasource-docker-tags",
       "key": "https://***.dkr.ecr.eu-west-2.amazonaws.com:/ecr-repo",
       "ttlMinutes": 30
```

Config set up:
```
{
  "branchPrefix": "renovate/",
  "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
  "onboarding": true,
  "platform": "github",
  "repositories":  [
      "org/repo"
  ],
  "helm-values": {
    "fileMatch": ["^manifests\\/api\\/cluster-test\\.yml$"],
    "matchStrings": [
      "tag: '(?<currentValue>.*?)'\n"
    ]
  },
  "hostRules": [
    {
      "hostType": "ecr",
      "matchHost": "https://0000000000.dkr.ecr.eu-west-2.amazonaws.com",
      "username": "AWS_ACCESS_KEY_ID",
      "password": "AWS_SECRET_ACCESS_KEY"
    }
  ]
}
```
Im Authenticating into AWS and ecr login using:
aws-actions/configure-aws-credentials@v1
aws-actions/amazon-ecr-login@v1

Github action step replaces the access keys secrets:

```
      - name: grab secrets
        run: |
          sed -i -e "s@AWS_ACCESS_KEY_ID@${{ secrets.AWS_ACCESS_KEY_ID }}@g" renovate-config.json
          sed -i -e "s@AWS_SECRET_ACCESS_KEY@${{ secrets.AWS_SECRET_ACCESS_KEY }}@g" renovate-config.json
```
I've tried this on a public docker image like crate and it works fine but soon as i test private repo like AWS ECR i get this. 
If i'm doing anything wrong in terms of set up  please can you point out or Any help on this would be great Thanks 🙏

**Comment by rarkins** [comment]
`renovate-config.json` is not a supported configuration file name. How are you using it?

**Comment by reitzig** [comment]
We're seeing the same error. We run Renovate in OpenShift and obtain AWS credentials (access key ID + secret access key) using a `CredentialsRequest` (which is minted and rotated regularly by an operator, or so I've been told).

On the CLI, we can reproduce the error from TRACE level by running `aws ecr get-authorization-token`:
```
User: arn:aws:iam::xys is not authorized to perform: ecr:GetAuthorizationToken on resource: * because no identity-based policy allows the ecr:GetAuthorizationToken action
```

However, we _can_ perform `aws ecr list-images` with the credentials we have!

We're still investigating whether something's off in our IAM setup, but is it possible that there are legitimate scenarios in which getting a short-lived authorization token is redundant and Renovate should skip skip that step? And instead use the provided credentials for its requests?
